crates/wasmtime/src/runtime/module.rs

```diff
@@ -381,6 +448,117 @@ impl Module {
+                    match relocation.target() {
+                        object::RelocationTarget::Symbol(symbol_idx) => {
+                            match file.symbol_by_index(symbol_idx) {
+                                Ok(symbol) => {
+                                    let addend =
+                                        symbol.address().wrapping_add(relocation.addend() as u64);
+                                    relocation.set_addend(addend as i64);
+                                }
+                                Err(_) => {
+                                    eprintln!(
+                                        "Relocation with invalid symbol for section {} at offset 0x{:08x}",
+                                        section.name().unwrap(),
+                                        offset
+                                    );
+                                }
+                            }
+                        }
+                        _ => {}
+                    }
+                    if relocations.insert(offset, relocation).is_some() {
+                        eprintln!(
+                            "Multiple relocations for section {} at offset 0x{:08x}",
+                            section.name().unwrap(),
+                            offset
+                        );
+                    }
+                }
+                _ => {
+                    eprintln!(
+                        "Unsupported relocation for section {} at offset 0x{:08x}",
+                        section.name().unwrap(),
+                        offset
+                    );
+                }
+            }
+        }
+    }
+
+    fn load_file_section<'input, 'arena, Endian: gimli::Endianity>(
+        id: gimli::SectionId,
+        file: &object::File<'input>,
+        endian: Endian,
+        is_dwo: bool,
+        arena_data: &'arena Arena<Cow<'input, [u8]>>,
+    ) -> Result<gimli::EndianSlice<'input, Endian>>
+    where
+        'arena: 'input,
+    {
+        let mut relocations = RelocationMap::default();
+        let name = if is_dwo {
+            id.dwo_name()
+        } else if file.format() == object::BinaryFormat::Xcoff {
+            id.xcoff_name()
+        } else {
+            Some(id.name())
+        };
+
+        let data = match name.and_then(|name| file.section_by_name(&name)) {
+            Some(ref section) => {
+                // DWO sections never have relocations, so don't bother.
+                if !is_dwo {
+                    Module::add_relocations(&mut relocations, file, section);
+                }
+                section.uncompressed_data()?
+            }
+            // Use a non-zero capacity so that `ReaderOffsetId`s are unique.
+            None => Cow::Owned(Vec::with_capacity(1)),
+        };
+        let data_ref = arena_data.alloc(data);
+        let reader = gimli::EndianSlice::new(data_ref, endian);
+        let section = reader;
+        Ok(section)
+    }
+
+    fn load_dwp<'a>(
```

Comment:
Would it be possible to thread through the `*.dwp` bytes through to the debug processing, but otherwise leave them unmodified before then? That'd help keep all the debuginfo processing centralized in one location rather than spread out.

Reply:
ok, let me see how that works out.

Reply:
I've passed the bytes through as low as module_environ.cs which is where I'm setting up the `debuginfo.dwarf_package`.  Is that what you had in mine, or having `debuginfo.dwarf_packages` as  bytes also?